src/components/Todo/Todo.tsx

```diff
@@ -0,0 +1,258 @@
+
+            newTodos[idx].isAwaitServer = false;
+
+            return newTodos;
+          });
+        });
+    };
+  };
+
+  const handleOnDeleteButtonTodo = (idx: number) => {
+    return () => {
+      refreshTodos(todos => {
+        const newTodos = [...todos];
+
+        newTodos[idx].isAwaitServer = true;
+
+        return newTodos;
+      });
+
+      deleteTodo(initialTodos[idx].id)
+        .then(() => {
+          refreshTodos(todos => {
+            const newTodos = todos.filter((_, i) => i !== idx);
+
+            return newTodos;
+          });
+        })
+        .catch(() => {
+          setErrorMsg('Unable to delete a todo');
+        });
+    };
+  };
+
+  const saveData = (idx: number) => {
+    return () => {
+      let indexTodoToBeUpdated: number;
+
+      refreshTodos(prev => {
+        indexTodoToBeUpdated = prev.findIndex(
+          e => e.id === initialTodos[idx].id,
+        );
+        const newPrev = [...prev];
+
+        newPrev[indexTodoToBeUpdated].isAwaitServer = true;
+
+        return newPrev;
+      });
+
+      if (valueInInputForEditTodo === initialTodos[idx].title) {
+        setTodoClickToEdit(-1);
+        setValueInInputForEditTodo('');
+
+        return;
+      }
+
+      if (valueInInputForEditTodo !== '') {
+        setRequestRunning(false);
+        updateTodo(initialTodos[idx].id, {
+          title: valueInInputForEditTodo.trim(),
+        })
+          .then(res => {
+            refreshTodos(prev => {
+              setTodoClickToEdit(-1);
+              const newPrev = [...prev];
+
+              newPrev[indexTodoToBeUpdated] = { ...res, isAwaitServer: false };
+
+              return newPrev;
+            });
+            setRequestRunning(true);
+          })
+          .catch(() => {
+            refreshTodos(prev => {
+              const newPrev = [...prev];
+
+              newPrev[indexTodoToBeUpdated].isAwaitServer = false;
+              setErrorMsg('Unable to update a todo');
+
+              return newPrev;
+            });
+            setRequestRunning(true);
+          });
+      } else {
+        deleteTodo(initialTodos[idx].id)
+          .then(() => {
+            refreshTodos(todos => {
+              const newTodos = todos.filter((_, i) => i !== idx);
+
+              return newTodos;
+            });
+            setRequestRunning(true);
+          })
+          .catch(() => {
+            refreshTodos(prev => {
+              const newPrev = [...prev];
+
+              newPrev[indexTodoToBeUpdated].isAwaitServer = false;
+              setErrorMsg('Unable to delete a todo');
+
+              return newPrev;
+            });
+            setRequestRunning(true);
+          });
+      }
+    };
+  };
+
+  const handleOnBlurEvent = (idx: number) => {
+    return () => {
+      if (requestRunning) {
+        saveData(idx)();
+      }
+    };
+  };
+
+  const handleClickToEdit = (idx: number) => {
+    return () => {
+      setTodoClickToEdit(initialTodos[idx].id);
```

Comment:
Instead of receiving the index, wouldn't it be easier to receive the Todo instead?